src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,19 +114,26 @@ private void checkHealth() {
     }
   }
 
-  public void stop() {
+  public synchronized void stop() {
```

Comment:
I think it is OK for stop() to be called before start(). In that case stop can just return and succeed immediately, right? If we have use-case in which this approach doesn't work, we can throw exception.